File: src/main/java/com/example/edumanager/entities/Aluno.java
package com.example.edumanager.entities;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Aluno {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    public String nome;

    public String CPF;

    public String email;

    public Date dataNascimento;

    public String matricula;

    public Date dataMatricula;

    public String telefone;

    // FK de turma
}

File: front/app/types/aluno.ts
export class Aluno{

    constructor(
        public id: number | null,
        public nome: string,
        public cpf: string,
        public email: string,
        public dataNascimento: string,
        public matricula: string,
        public dataMatricula: string,
        public status: string,
        public situacao: string)
        {        

    }
}